src/app/Main/Body/QueryPage/ChangeFilter.service.ts

```diff
@@ -1,11 +1,9 @@
-import { ValueType } from "react-select";
-
 export function changeFilter(
-    selectedOption: ValueType<{value: string, label: string}>
+    selectedOption: { value: string; label: string }[]
 ): string[] {
-    if (Array.isArray(selectedOption)) {
+    if (selectedOption !== undefined) {

```

Comment:
same here

Reply:
This function has been changed to export function export function chooseSelectedFiltersService. Can you please look into these functions again and say if changes are still needed?

Reply:
looks good, but service is kind of obsolete now and can be replaced by:
`selectedFilters = selectedOption  === null ? [] : selectedOption.map(option => option.value)`
or completely removed if your selector component returns: `onSelectionChange(options: string[])` with options = [] for no selection.